src/Components/Pet.js

```diff
@@ -1,55 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../Styles/Pet.scss';
+import { useQuery, gql } from '@apollo/client';
 import Header from './Header';
+import { useParams } from 'react-router-dom';
+import OwnerInformation from './OwnerInformation';
+import PetInformation from './PetInformation';
 
 
+const GET_SINGLE_PET = gql`
+query getPetById($id: ID!) {
+  getPetById(id: $id) 
+  {
+    id
+    name
+    age
+    gender
+    description
+    species
+    ownerStory
+    ownerEmail
+    ownerName
+    applications {
+      name
+      email
+      description
+    }
+  }
+}
+`
+
 const Pet = () => {
+  const [showOwnerInfo, setOwnerInfo] = useState(false);
+  
+  let { id } = useParams();
+  const { loading, error, data } = useQuery(GET_SINGLE_PET, {variables: {id}})
+
+  if (loading) {
+    return <p>Loading...</p>
+  }
+  if (error) {
+    return <p>{error}</p>
+  }
+
+
+  let currentView;
```

Comment:
I really like this logic that allows us to change what components show up on the card based on what the user needs to see, and that we made the pet info it's own component we could pass down the function for switching the page to.